### PR TITLE
reconstruct dbt adapter in DbtCLIInvocation

### PR DESCRIPTION
**Pull request by prha**
## Summary & Motivation
Enables the extraction of some of the adapter profile information for reuse.

## How I Tested These Changes
BK

### CODE REVIEW

**Review by rexledesma — COMMENTED** [review]
Just passing by here -- some other reference code that might be of use as you go down this route to construct the adapter:

- [`sqlmesh`'s implementation](https://github.com/TobikoData/sqlmesh/blob/1c5212a8fafa10f0539d9bceb4e5b7b5d7299ade/sqlmesh/dbt/manifest.py#L239-L286
)
- [`dbt-fal`'s implementation](https://github.com/fal-ai/dbt-fal/blob/cbdfc5dabad46c29b3e89d04143caf4a16fc14de/projects/adapter/src/fal/dbt/integration/lib.py#L90-L105)

**Comment by prha** [comment]
> some other reference code that might be of use 

Yeah, these impls look a lot cleaner.  I guess I'm not certain enough of the interop between the CLI args of the `DbtCliResource` and the expected invocation of the runtime config to know whether it's safe to invoke without the `make_context` call.

File: python_modules/libraries/dagster-dbt/dagster_dbt/utils.py [hunk 1]
```diff
@@ -310,3 +310,13 @@ def get_dbt_resource_props_by_dbt_unique_id_from_manifest(
         **manifest["metrics"],
         **manifest.get("semantic_models", {}),
     }
+
+
+def _set_flag_attr(key, value):
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/utils.py:315]: nit, so we don't have to iterate through all of the global flags twice
```suggestion
def _set_flags(kvs: Sequence[Tuple[str, Any]]):
```

File: python_modules/libraries/dagster-dbt/dagster_dbt/utils.py [hunk 2]
```diff
@@ -285,8 +285,8 @@ def __getattr__(self, item):
     graph = graph_selector.Graph(DiGraph(incoming_graph_data=child_map))
 
     # create a parsed selection from the select string
-    setattr(GLOBAL_FLAGS, "INDIRECT_SELECTION", IndirectSelection.Eager)
-    setattr(GLOBAL_FLAGS, "WARN_ERROR", True)
+    _set_flag_attr("INDIRECT_SELECTION", IndirectSelection.Eager)
+    _set_flag_attr("WARN_ERROR", True)
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/utils.py:289]:
```suggestion
    _set_flags([("INDIRECT_SELECTION", IndirectSelection.Eager), ("WARN_ERROR", True)])
```

**Review by rexledesma — COMMENTED** [review]
A simple test confirming that we get the duckdb adapter when we execute this in our test suite would be great

**Review by rexledesma — APPROVED** [review]
Nice. Do we need to pass `RuntimeConfig` as well? Not sure if that's needed in insights.

Would also like to see an end-to-end example of insights working on Bigquery before merging, just to be certain

**Pull request merged**